controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -173,20 +176,35 @@ func (w *Workload) getConfigMap(ctx context.Context, configMap ctrlclient.Object
 }
 
 // UpdateImageRepositoryInKubeadmConfigMap updates the image repository in the kubeadm config map.
-func (w *Workload) UpdateImageRepositoryInKubeadmConfigMap(ctx context.Context, imageRepository string, version semver.Version) error {
-	return w.updateClusterConfiguration(ctx, func(c *bootstrapv1.ClusterConfiguration) {
+func (w *Workload) UpdateImageRepositoryInKubeadmConfigMap(imageRepository string) func(*bootstrapv1.ClusterConfiguration, *[]string) {
+	return func(c *bootstrapv1.ClusterConfiguration, updatedKeys *[]string) {
 		if imageRepository == "" {
 			return
 		}
+
 		c.ImageRepository = imageRepository
-	}, version)
+		*updatedKeys = append(*updatedKeys, "imageRepository")
+	}
+}
+
+// UpdateFeatureGatesInKubeadmConfigMap updates the feature gates in the kubeadm config map.
+func (w *Workload) UpdateFeatureGatesInKubeadmConfigMap(featureGates map[string]bool) func(*bootstrapv1.ClusterConfiguration, *[]string) {
+	return func(c *bootstrapv1.ClusterConfiguration, updatedKeys *[]string) {
+		if featureGates == nil {
+			return
+		}
```

Comment:
if featureGates is nil, we should maybe unset it or set an empty map[string]bool?

Reply:
good catch